biopet/biopet.wdl

```diff
@@ -205,54 +205,55 @@ task FastqSync {
 task ReorderGlobbedScatters {
     input {
         Array[File]+ scatters
-        String scatterDir
+        # Should not be changed from the main pipeline. As it should not influence results.
+        String dockerTag = "3.6"
     }
 
     command <<<
+       set -e
+       # Copy all the scatter files to the CWD so the output matches paths in
+       # the cwd.
+       for file in ~{sep=" " scatters}
+          do cp $file .
+       done
        python << CODE
        from os.path import basename
        scatters = ['~{sep="','" scatters}']
        splitext = [basename(x).split(".") for x in scatters]
        splitnum = [x.split("-") + [y] for x,y in splitext]
        ordered = sorted(splitnum, key=lambda x: int(x[1]))
-       merged = ["~{scatterDir}/{}-{}.{}".format(x[0],x[1],x[2]) for x in ordered]
+       merged = ["{}-{}.{}".format(x[0],x[1],x[2]) for x in ordered]
        for x in merged:
            print(x)
        CODE
     >>>
 
     output {
-        Array[String] reorderedScatters = read_lines(stdout())
+        Array[File] reorderedScatters = read_lines(stdout())
     }
 
     runtime {
+        docker: "python:" + dockerTag
         memory: 1
     }
 }
 
 task ScatterRegions {
     input {
-        String? preCommand
         Reference reference
-        String outputDirPath
-        File? toolJar
         Int? scatterSize
         File? regions
         Boolean notSplitContigs = false
-
         Int memory = 4
         Float memoryMultiplier = 3.0
+        String dockerTag = "0.2--0"
     }
-
-    String toolCommand = if defined(toolJar)
-        then "java -Xmx" + memory + "G -jar " +toolJar
-        else "biopet-scatterregions -Xmx" + memory + "G"
+    String outputDirPath = "scatters"
```

Comment:
Why move this out of the inputs?

Reply:
It was a lot of hassle getting stuff to work correctly with the file localization. But I will return this to normal and see if I can get that working again.

Reply:
I think it's fine to keep like this, but then this variable is completely obsolete

Reply:
No this will never work.
Glob works by linking. And linking between paths inside the container and outside will not work. So no links will appear and the glob will return an empty list.

Reply:
Hmm okay, so in that case the variable can be removed altogether?

Reply:
Well. Since it is used in 3 places it is nice to keep it in a variable.